include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -15,6 +15,7 @@ namespace hlsl
 namespace spirv
 {
 
+#ifdef __HLSL_VERSION
```

Comment:
you probably want to move this up so `nbl::hlsl::spirv` isn't even a namespace for C++